<ui version="4.0">
 <author/>
 <comment/>
 <exportmacro/>
 <class>SphereBot</class>
 <widget class="QMainWindow" name="SphereBot">
 <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1050</width>
    <height>655</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1040</width>
    <height>655</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1050</width>
    <height>655</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QGroupBox" name="controllBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>551</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>8</x>
       <y>0</y>
       <width>541</width>
       <height>221</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_15">
        <item>
         <widget class="QPushButton" name="setDiameterButton">
          <property name="text">
           <string>set Diameter</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QSlider" name="diameterSlider">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>120</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QSpinBox" name="diameterpinBox">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>120</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Obj. Diameter</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_13">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <spacer name="verticalSpacer_5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QSlider" name="servoFeedrateSlider">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>6000</number>
            </property>
            <property name="value">
             <number>3000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_6">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSpinBox" name="FeedratespinBox">
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>6000</number>
          </property>
          <property name="value">
           <number>3000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Feedrate</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_12">
          <item>
           <spacer name="verticalSpacer_7">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QSlider" name="servoSlider">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="minimum">
             <number>23</number>
            </property>
            <property name="maximum">
             <number>120</number>
            </property>
            <property name="value">
             <number>35</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="invertedAppearance">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_8">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSpinBox" name="servospinBox">
          <property name="minimum">
           <number>23</number>
          </property>
          <property name="value">
           <number>35</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Servo</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QSlider" name="penSlider">
              <property name="minimum">
               <number>-112</number>
              </property>
              <property name="maximum">
               <number>112</number>
              </property>
              <property name="singleStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>0</number>
              </property>
              <property name="sliderPosition">
               <number>0</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="invertedAppearance">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="penslider">
                <property name="text">
                 <string>Pen Rotation Y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="penRotationBox">
                <property name="minimum">
                 <number>-112</number>
                </property>
                <property name="maximum">
                 <number>112</number>
                </property>
                <property name="singleStep">
                 <number>1</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QSlider" name="eggSlider">
            <property name="minimum">
             <number>-450</number>
            </property>
            <property name="maximum">
             <number>450</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="sliderPosition">
             <number>0</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="penslider_2">
              <property name="text">
               <string>Egg Rotation X</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="eggRotationBox">
              <property name="minimum">
               <number>-450</number>
              </property>
              <property name="maximum">
               <number>450</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>275</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="connectButton">
       <property name="text">
        <string>Connect</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="portBox"/>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_9"/>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_10"/>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>290</y>
      <width>541</width>
      <height>211</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_11">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLineEdit" name="sendString"/>
       </item>
       <item>
        <widget class="QPushButton" name="sendButton">
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Transceived Data</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="txList">
           <property name="autoScroll">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::NoSelection</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Received Data</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="rxList"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>10</y>
      <width>461</width>
      <height>461</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_12">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8"/>
       </item>
       <item>
        <widget class="QLabel" name="fileName">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QPushButton" name="undoButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Undo</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="redoButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Redo</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="fileTextEdit"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QPushButton" name="loadFileButton">
           <property name="text">
            <string>Load File</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="saveFileButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Save File</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="sendFileButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Send File</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="restartButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Restart</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QProgressBar" name="fileSendProgressBar">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>510</y>
      <width>1011</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_14"/>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_16"/>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_17"/>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_15"/>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>controllBox</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>graphicsView</zorder>
   <zorder>layoutWidget</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>diameterSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>diameterpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>71</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>diameterpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>diameterSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>66</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>servospinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>servoSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>272</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>servoSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>servospinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>272</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>penRotationBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>penSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>553</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>554</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>penSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>penRotationBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>554</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>498</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eggRotationBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>eggSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eggSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>eggRotationBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>servoFeedrateSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FeedratespinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>206</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FeedratespinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>servoFeedrateSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>178</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
